src/slice-chop.c

```diff
@@ -202,7 +203,8 @@ static SEXP chop(SEXP x, SEXP indices, struct vctrs_chop_info info) {
       UNPROTECT(1);
     }
 
-    elt = vec_restore(elt, x, info.restore_size);
+    const enum vctrs_ownership ownership = vec_ownership(elt);
```

Comment:
Probably no need to declare it const since it's passed by value anyway? Maybe inline this in the restore call?